### PR TITLE
Replace black by Ruff Formatter

### PR DESCRIPTION
**Pull request by hussein-awala**
This PR replaces black with Ruff Formatter, which is 30x faster (https://astral.sh/blog/the-ruff-formatter). It also removes the `--skip-string-normalization` config to replace single quotes with double quotes.

related: apache/iceberg#8294

### CODE REVIEW

**Comment by rdblue** [comment]
Looks fine overall, but it seems like too many changes with string normalization. Why force string normalization? That's going to cause a ton of pull requests to fail formatting validation.

File: .pre-commit-config.yaml
Comment (Fokko) [inline .pre-commit-config.yaml:32]: Does it come with the new version? I don't see any related config (for example, line length)

**Comment by hussein-awala** [comment]
> Looks fine overall, but it seems like too many changes with string normalization. Why force string normalization? That's going to cause a ton of pull requests to fail formatting validation.

It is not supported yet; we can wait for https://github.com/astral-sh/ruff/pull/8822 before merging.

File: .pre-commit-config.yaml
Comment (hussein-awala) [reply .pre-commit-config.yaml:32]: yes it was introduced in https://github.com/astral-sh/ruff/releases/tag/v0.0.290, but I used the newer version to get some patches.

**Comment by Fokko** [comment]
@hussein-awala the PR has been merged :)

**Comment by hussein-awala** [comment]
> @hussein-awala the PR has been merged :)

I was waiting for the release, the merged PR was released 3 days ago in [v0.1.8](https://github.com/astral-sh/ruff/releases/tag/v0.1.8) as a preview feature, for that I had to add `--preview` argument for `ruff` and `ruff-format`.

> Looks fine overall, but it seems like too many changes with string normalization. Why force string normalization? That's going to cause a ton of pull requests to fail formatting validation.

@rdblue could you check it now after changing `quote-style` to `preserve`?

**Review by Fokko — APPROVED** [review]
It looks like there are still differences, but I like them since it makes the code a bit more dense, without sacrificing readability.

**Comment by rdblue** [comment]
Looks good to me now. Thanks, @hussein-awala!

**Comment by hussein-awala** [comment]
I just merged main and fixed the conflicts, it should be ready to merge.

**Review by Fokko — APPROVED** [review]
Thanks again @hussein-awala for the contribution, and @rdblue for the review 🙌

**Comment by hussein-awala** [comment]
We need to merge https://github.com/apache/iceberg-python/pull/253 to fix the failed tests in main.

**Pull request merged**
